From a9288aa6c96fdd465d591e4dcd2d29a996a12f75 Mon Sep 17 00:00:00 2001
From: Paul Scheffler <paulsc@iis.ee.ethz.ch>
Date: Wed, 15 Feb 2023 18:05:40 +0100
Subject: [PATCH] treewide: Further fixes

---
 rtl/i2c_fsm.sv | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/rtl/i2c_fsm.sv b/rtl/i2c_fsm.sv
index ad7246bf4..ff019408f 100644
--- a/rtl/i2c_fsm.sv
+++ b/rtl/i2c_fsm.sv
@@ -4,6 +4,8 @@
 //
 // Description: I2C finite state machine
 
+`include "common_cells/assertions.svh"
+
 module i2c_fsm import i2c_pkg::*;
 #(
   parameter int FifoDepth = 64,
-- 
2.16.5

