From d635f031f9bfb26e5f48ac5cd12c5ff332af8ac4 Mon Sep 17 00:00:00 2001
From: Tim Fischer <fischeti@iis.ee.ethz.ch>
Date: Fri, 13 Jun 2025 14:57:06 +0200
Subject: [PATCH] gpio: Replace non-resettable FFs

---
 rtl/gpio.sv | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/rtl/gpio.sv b/rtl/gpio.sv
index 3c2b948b5b..ac75504a0f 100644
--- a/rtl/gpio.sv
+++ b/rtl/gpio.sv
@@ -5,6 +5,7 @@
 // General Purpose Input/Output module
 
 `include "common_cells/assertions.svh"
+`include "common_cells/registers.svh"
 
 module gpio
   import gpio_reg_pkg::*;
@@ -109,9 +110,7 @@ module gpio
   end
 
   logic [31:0] data_in_q;
-  always_ff @(posedge clk_i) begin
-    data_in_q <= data_in_d;
-  end
+  `FF(data_in_q, data_in_d, '0, clk_i, rst_ni)
 
   logic [31:0] event_intr_rise, event_intr_fall, event_intr_actlow, event_intr_acthigh;
   logic [31:0] event_intr_combined;
-- 
2.42.0

