From 55ad9fdb2e514fbfc0641a119fa0f62352e3a65d Mon Sep 17 00:00:00 2001
From: Paul Scheffler <paulsc@iis.ee.ethz.ch>
Date: Wed, 15 Feb 2023 15:37:56 +0100
Subject: [PATCH] treewide: Additional rebase fixes

---
 rtl/gpio.sv | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/rtl/gpio.sv b/rtl/gpio.sv
index 27c7df0eb..3c2b948b5 100644
--- a/rtl/gpio.sv
+++ b/rtl/gpio.sv
@@ -4,7 +4,6 @@
 //
 // General Purpose Input/Output module
 
-`include "prim_assert.sv"
 `include "common_cells/assertions.svh"
 
 module gpio
@@ -164,6 +163,4 @@ module gpio
   `ASSERT_KNOWN(CioGpioEnOKnown, cio_gpio_en_o)
   `ASSERT_KNOWN(CioGpioOKnown, cio_gpio_o)
 
-  // Alert assertions for reg_we onehot check
-  `ASSERT_PRIM_REG_WE_ONEHOT_ERROR_TRIGGER_ALERT(RegWeOnehotCheck_A, u_reg, alert_tx_o[0])
 endmodule
-- 
2.16.5

